chromadb/api/models/Collection.py

```diff
@@ -273,6 +273,54 @@ def update(
 
         self._client._update(self.name, ids, embeddings, metadatas, documents)
 
+    def upsert(
```

Comment:
The vast bulk of this method is validations, and AFAIKT the validations for an upsert should be the same as for an insert. Any way to factor out the duplicated validation code?